lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -517,8 +526,18 @@ def rescore_problem_module_state(xmodule_instance_args, module_descriptor, stude
             msg = "Specified problem does not support rescoring."
             raise UpdateProblemModuleStateError(msg)
 
+        # set the tracking info before this call, because
+        # it makes downstream calls that create events
+        create_new_event_transaction_id()
```

Comment:
Just have `create_new_event_transaction_id` return the id instead of calling `get_event_transaction_id` again below.